docs/content/docs/release_guide.md

```diff
@@ -37,55 +39,45 @@ Push the release branch to the main repo (not fork), identified here by the `ups
+Create a new commit on the new branch to update the hardcoded version information in the following locations:
 
-The pull request must be reviewed and merged before proceeding to the next step.
+* The container image tag in [charts/osm/values.yaml](https://github.com/openservicemesh/osm/tree/main/charts/osm/values.yaml) to point to the new release
+* The chart and app version in [charts/osm/Chart.yaml](https://github.com/openservicemesh/osm/tree/main/charts/osm/Chart.yaml) to point to the new release
+* The default osm-controller image tag in [osm cli](https://github.com/openservicemesh/osm/blob/main/cmd/cli/install.go) to point to the new release
+* The Helm chart [README.md](https://github.com/openservicemesh/osm/blob/main/charts/osm/README.md)
+  - Necessary changes should be made automatically by running `make chart-readme`
+
+Once patches and version information have been updated on a new branch off of the release branch, create a pull request from the new branch to the release branch. Proceed to the next step once the pull request is approved and merged.
 
 ## Create and push a Git tag
 
-Once the release is ready to be published, create and push a Git tag from the release branch in
+Ensure your local copy of the release branch has the latest changes from the PR merged above.
+
+Once the release is ready to be published, create and push a Git tag from the release branch to
 the main repo (not fork), identified here by the `upstream` remote.
 
 ```console
 $ export RELEASE_VERSION=<release-version> # ex: export RELEASE_VERSION=v0.4.0
-$ git tag -a "$RELEASE_VERSION" -m "<add description here>"
+$ git tag "$RELEASE_VERSION"
 $ git push upstream "$RELEASE_VERSION"
 ```
 
 A [GitHub Action](https://github.com/openservicemesh/osm/blob/main/.github/workflows/release.yml) is triggered when the tag is pushed.
 It will build the CLI binaries, publish a new GitHub release,
-upload the packaged binaries and checksums as release assets,
-and build and push Docker images for OSM and the demo to the
-[`openservicemesh` organization](https://hub.docker.com/u/openservicemesh) on Docker Hub.
+upload the packaged binaries and checksums as release assets, build and push Docker images for OSM and the demo to the
+[`openservicemesh` organization](https://hub.docker.com/u/openservicemesh) on Docker Hub, and publish the Helm chart to the repo hosted at https://openservicemesh.github.io/osm.
 
 ## Add release notes
 
 In the description section of the new release, add information about feature additions, bug fixes,
 and any other administrative tasks completed on the repository.
 
-## Update README chart with helm-docs
-
-Chart's README is autogenerated with [helm-docs](https://github.com/norwoodj/helm-docs). Once versioning has been addressed
-under `charts/osm/Chart.yaml`, update the README with the following command:
-
-```
-make chart-readme
-```
-
-## Create a Helm chart release
```

Comment:
Is this now being done in the same github action triggered by the git tag?

Reply:
Different action (to make it easy to separate in the future), but both will now get kicked off by the same `vx.x.x` tag.